capstone-rs/src/instruction.rs

```diff
@@ -178,21 +172,8 @@ macro_rules! impl_SliceIterator_wrapper {
     }
 }
 
-/// An iterator over the instructions returned by disasm
-///
-/// This is currently the only supported interface for reading them.
-pub struct InstructionIterator<'a>(slice::Iter<'a, cs_insn>);
-
-impl_SliceIterator_wrapper!(
-    impl<'a> Iterator for InstructionIterator<'a> {
-        type Item = Insn<'a>;
-        [
-            |x| Insn { insn: *x, _marker: PhantomData }
-        ]
-    }
-);
-
 /// A wrapper for the raw capstone-sys instruction
+#[derive(Clone)]
```

Comment:
I need to think more if this is sound.
`cs_insn` has the `detail: *mut cs_detail` field.

Reply:
This should be safe since the lifetime of `Insn` is tied to `Instructions` and that's what frees the pointer, but I might be wrong.

Reply:
I think you are right. I *think* the `Clone` should be safe. But, is it useful?